pyproject.toml

```diff
@@ -1,6 +1,219 @@
-# build-system taken from https://github.com/pypa/setuptools_scm/#pyprojecttoml-usage
+[project]
+requires-python = ">=3.7"
+
+name        = "armory-testbed"
+description = "Adversarial Robustness Test Bed"
+dynamic     = [ "version" ]
+license     = "MIT"
+authors = [
+  { name = "Two Six Technologies", email = "armory@twosixtech.com" },
+]
+
+classifiers = [
+  "Development Status :: 3 - Alpha",
+  "License :: OSI Approved :: MIT License",
+  "Environment :: Console",
+  "Natural Language :: English",
+  "Intended Audience :: Developers",
+  "Intended Audience :: Education",
+  "Intended Audience :: Science/Research",
+  "Programming Language :: Python :: 3.7",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Topic :: Scientific/Engineering :: Artificial Intelligence",
+]
+
+# the minimal set needed to run armory.main (launcher)
+dependencies = [
+    "wheel",
+    "setuptools_scm",
+    "docker",
+    "jsonschema",
+    "loguru",
+    "pytest",
+    "PyYAML",
+    "requests",
+    "toml",
+]
+
+[project.urls]
+Source        = "https://github.com/twosixlabs/armory"
+Documentation = "https://armory.readthedocs.io/en/latest/"
+
+[project.scripts]
+armory = "armory.__main__:main"
+
+
+# please see docs/getting_started.md#the-armory-flavors for how these
+# optional dependencies work
+[project.optional-dependencies]
+developer =[
+    "hatch", # build tools
+    "bandit[toml]", # code scanning
+    "black", # formatting
+    "black[jupyter]",
+    "flake8",
+    "mkdocs",
+    "Markdown",
+    "yamllint",
+    "tox", # testing
+    "Jinja2",
+    "podman",
+    "hydra-core", # configuration
+    "omegaconf",
+]
+
+engine = [
+    "adversarial-robustness-toolbox==1.12.1",
+    "python-levenshtein",
+    "Pillow == 9.2.0",           # Data dependencies
+    "boto3 == 1.24.72",          # Needed for armory.data.utils
+    "botocore == 1.27.77",       # Needed for armory.data.utils
+    "ffmpeg-python == 0.2.0",    # Needed for armory.utils.export
+    "opencv-python == 4.6.0.66"  # Needed for CARLA baseline scenario
+]
+
+datasets = [
+    "tensorflow-datasets == 4.6.0",
+    "apache-beam >= 2.22.0",
+    "pydub >= 0.24.1",
+]
+
+math = [
+    "numpy >= 1.21.0",
+    "scipy >= 1.3.2",
+    "scikit-learn >= 1.0.0",
+    "matplotlib == 3.5.3",
+]
+
+pytorch = [
+    "armory-testbed[engine,datasets,math]",
+    "sox == 1.4.1",
+    "librosa == 0.9.2",
+    "google-cloud-storage == 2.5.0",
+    "torch == 1.12.1",
+    "torchvision == 0.13.1",
+    "torchaudio == 0.12.1",
+    "torchmetrics == 0.9.3",
+    "transformers",
+    "pytorch-lightning == 1.4.9",
+    "hydra-core == 1.2.0",
+    "hydra-configs-pytorch-lightning @ git+https://github.com/romesco/hydra-lightning/#subdirectory=hydra-configs-pytorch-lightning",
```

Comment:
`pytorch` should only include `torch`, `torchaudio`, and `torchvision`. The rest of these should be pushed to `deepspeech`